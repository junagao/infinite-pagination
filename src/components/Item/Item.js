import React from 'react'
import { string } from 'prop-types'
import { Button } from 'components'

const Item = ({ name }) => (
  <li>
    <Button text={name}>{name}</Button>
  </li>
)

Item.propTypes = {
  name: string.isRequired,
}

export default Item
