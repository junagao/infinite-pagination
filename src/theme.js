const theme = {
  colors: {
    background: '#fff',
    primary: '#111',
    text: { body: '#111', hover: '#f0f0f0' },
    disabled: '#ddd',
    error: '#BD271E',
  },
  fontSizes: {
    xl: '4rem',
    lg: '2rem',
    md: '1.5rem',
    sm: '1rem',
    xs: '0.75rem',
  },
  space: [
    0,
    '0.25rem',
    '0.5rem',
    '0.75rem',
    '1rem',
    '1.5rem',
    '2rem',
    '4rem',
    '6rem',
  ],
  borders: {
    none: 'none',
    thin: '1px solid',
  },
  radii: {
    none: 0,
    base: '0.25em',
    round: '99999em',
  },
}

export default theme
