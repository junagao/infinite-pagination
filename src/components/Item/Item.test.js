import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'emotion-theming'
import theme from '../../theme'
import Item from './Item'

describe('Item', () => {
  test('matches snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Item id="yt302" name="item-1" selected onSelect={jest.fn()} />
      </ThemeProvider>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders one li item with one instance of button', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Item id="yt302" name="item-1" selected onSelect={jest.fn()} />
      </ThemeProvider>,
    )
    const button = getByTestId('button')
    expect(getByTestId('list-item')).toContainElement(button)
  })
})
