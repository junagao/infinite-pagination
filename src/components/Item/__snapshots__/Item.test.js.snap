// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item matches snapshot 1`] = `
.emotion-0 {
  border: none;
  color: #ddd;
  cursor: default;
  font-size: 4rem;
  outline: none;
  padding: 1rem 0;
  pointer-events: none;
  -webkit-transition: all 0.2s cubic-bezier(0.645,0.045,0.355,1) 0s;
  transition: all 0.2s cubic-bezier(0.645,0.045,0.355,1) 0s;
  width: 100vw;
}

.emotion-0:hover {
  color: #ddd;
  background-color: #fff;
}

<li
  data-testid="list-item"
>
  <button
    class="emotion-0 emotion-1"
    data-testid="button"
    disabled=""
    type="button"
  >
    item-1
  </button>
</li>
`;
